pkgs/build-support/docker/default.nix

```diff
@@ -110,8 +117,17 @@ rec {
     , tlsVerify ? true
 
     , name ? fixName "docker-image-${finalImageName}-${finalImageTag}.tar"
+
+    # These parameters are no longer needed
+    , os ? null
+    , arch ? null
+    , variant ? null
     }:
 
+    assert os != null -> throw "`os` parameter is no longer needed by `pullImage` function";
+    assert arch != null -> throw "`arch` parameter is no longer needed by `pullImage` function";
+    assert variant != null -> throw "`variant` parameter is no longer needed by `pullImage` function";
```

Comment:
I may have been too quick to dismiss the `arch` and `variant` as unnecessary, because a base image may be more general than the platform for which you want to build your own things. Evaluating an extra cross nixpkgs seems a bit disproportionate for this use case.

Reply:
Well, I think that most people who are building docker image with nix already have their app tightly integrated in the nix ecosystem and so, that must evaluate a cross nixpkgs to compile their app for other platforms.
Also has you said in the original discussion about this change, the other functions don't work like that. If we revert this change, I think we should also edit the other function to streamline the experience with `dockerTools`, since the other functions require evaluating an extra cross nixpkgs to be able to pick the right arch / variant for the wanted system, so in nearly every case, if you want to cross-compile a container you will need to evaluate an extra cross nixpkgs .

Reply:
Well, I think that most people who are building docker image with nix already have their app tightly integrated in the nix ecosystem and so they must evaluate a cross nixpkgs to compile their app for other platforms.
Has you said in the original discussion about this change, the others functions don't work like that. If we revert this change, I think we should also edit the others functions to streamline the experience with dockerTools, creating a lot of breaking changes.
Moreover, the others functions require evaluating an extra cross nixpkgs to be able to pick the right arch / variant on the wanted system for which the container needs to run on (the value of these parameters depend on the output of the nix golang package that itself depends on arch for which nixpkgs is being imported), so in nearly every case, if you want to cross-compile a container you will need to evaluate an extra cross nixpkgs (except if you use remote builder for each arch).
At least, doesn't this break the nix philosophy where we already have something to specify platforms specific things, does it ?

_Sorry for the long answer, but it seems necessary, even if I'm not 100% sure to have correctly understood your previous message :grimacing:_

Reply:
I agree that it is not an essential feature.

You can remove `variant` from the parameter list and asserts here, because it didn't exist before your pr.

For the other two, you should add to the message what else to do, and you could link to the manual.

Removing `arch` is still somewhat of an experiment. I hope we can make this simplification without causing much trouble.

Reply:
Ok, I'll work on this and all the remaining comments in the PR over the weekend. I will do my best to help you and make this change as smooth as possible.